src/components/ArticleVote.jsx

```diff
@@ -1,37 +1,33 @@
 import IconButton from '@mui/material/IconButton'
 import ThumbUpOffAltIcon from '@mui/icons-material/ThumbUpOffAlt';
 import ThumbUpAltIcon from '@mui/icons-material/ThumbUpAlt';
-import { useState, useEffect } from 'react';
+import { useState } from 'react';
 import {updateArticleVote} from './api'
 
 // TODO: after user login implemented, prevent user voting multiple times across different sessions
 
 function ArticleVote({article}) {
     
-    const [votes, setVotes] = useState(article.voteCount);
-    const [incrVote, setIncrVote] = useState(0);
+    const [votesCount, setVotesCount] = useState(article.voteCount);
+    const [incrVote, setIncrVote] = useState(-1);
     const [errMessage, setErrMessage] = useState('');
 
-    useEffect(() => {
-        if (incrVote !== 0) {
-            updateArticleVote(article.id, incrVote)
-                .then((articleData) => {
-                    setVotes(articleData.votes);
-                })
-                .catch((err) => {
-                    const errMsg = err.msg
-                        ? err.msg
-                        : 'An unknown error occurred';
-                    setErrMessage(errMsg);
-                });
-        }
-    }, [article, incrVote]);
-
     function handleClick() {
-        const incrVoteVal = incrVote <= 0
+        const incrVoteVal = incrVote === -1
             ? 1
             : -1;
+        setVotesCount(article.voteCount + incrVoteVal) // optimistic rendering
         setIncrVote(incrVoteVal);
+        updateArticleVote(article.id, incrVoteVal)
+            .then((articleData) => {
+                setVotesCount(articleData.votes);
+            })
+            .catch((err) => {
+                const errMsg = err.msg
+                    ? err.msg
+                    : 'An unknown error occurred';
+                setErrMessage(errMsg);
+            });
```

Comment:
Really nice, works optimistically as you are setting the state first and then performing the api request. 

AND combining this functionality into one function, for both downvote and upvote is very clean. Dealing with the error too. Great work!